File: mobile-app/lib/services/tags_service.dart
/// Service pour gérer les tags
import '../services/api_service.dart';
import 'package:dio/dio.dart';

class TagsService {
  final ApiService _apiService = ApiService();

  /// Lister tous les tags
  Future<Response> listTags() async {
    return await _apiService.get('/tags');
  }

  /// Créer un tag
  Future<Response> createTag(String name, {String? color}) async {
    return await _apiService.post('/tags', data: {
      'name': name,
      if (color != null) 'color': color,
    });
  }

  /// Ajouter un tag à un fichier
  Future<Response> addTagToFile(String fileId, String tagId) async {
    return await _apiService.post('/tags/$tagId/files/$fileId');
  }

  /// Retirer un tag d'un fichier
  Future<Response> removeTagFromFile(String fileId, String tagId) async {
    return await _apiService.delete('/tags/$tagId/files/$fileId');
  }

  /// Obtenir les tags d'un fichier
  Future<Response> getFileTags(String fileId) async {
    return await _apiService.get('/files/$fileId/tags');
  }

  /// Rechercher des fichiers par tag
  Future<Response> searchByTag(String tagId) async {
    return await _apiService.get('/tags/$tagId/files');
  }

  /// Supprimer un tag
  Future<Response> deleteTag(String tagId) async {
    return await _apiService.delete('/tags/$tagId');
  }
}

File: frontend-web/src/services/tagsService.js
/**
 * Service pour gérer les tags
 */
import apiClient from './api';

export const tagsService = {
  /**
   * Créer un tag
   */
  async createTag(name, color = '#2196F3') {
    const response = await apiClient.post('/tags', { name, color });
    return response.data;
  },

  /**
   * Lister les tags
   */
  async listTags() {
    const response = await apiClient.get('/tags');
    return response.data;
  },

  /**
   * Mettre à jour un tag
   */
  async updateTag(tagId, updates) {
    const response = await apiClient.patch(`/tags/${tagId}`, updates);
    return response.data;
  },

  /**
   * Supprimer un tag
   */
  async deleteTag(tagId) {
    const response = await apiClient.delete(`/tags/${tagId}`);
    return response.data;
  },

  /**
   * Ajouter des tags à une ressource
   */
  async addTagsToResource(resourceId, tagIds, resourceType) {
    const response = await apiClient.post(`/tags/resources/${resourceId}/add`, {
      tag_ids: tagIds,
      resource_type: resourceType,
    });
    return response.data;
  },

  /**
   * Retirer des tags d'une ressource
   */
  async removeTagsFromResource(resourceId, tagIds, resourceType) {
    const response = await apiClient.post(`/tags/resources/${resourceId}/remove`, {
      tag_ids: tagIds,
      resource_type: resourceType,
    });
    return response.data;
  },

  /**
   * Obtenir les ressources d'un tag
   */
  async getResourcesByTag(tagId) {
    const response = await apiClient.get(`/tags/${tagId}/resources`);
    return response.data;
  },
};





